python_modules/dagster/dagster/_grpc/client.py

```diff
@@ -376,6 +376,16 @@ def external_schedule_execution(
             ExternalScheduleExecutionArgs,
         )
 
+        # The timeout for the schedule can be defined in one of three ways.
+        #   1. By the default grpc timeout
+        #   2. By the DEFAULT_SCHEDULE_GRPC_TIMEOUT environment variable
+        #   3. By the client.
+        # The default timeout argument of this function takes the maximum of (1) and
+        # (2), while
+        # the client may pass a timeout argument via the
+        # `sensor_execution_args` object. If the timeout is passed from the client, we use that value irrespective of what the other timeout values may be set to.
+        timeout = external_schedule_execution_args.timeout or timeout
```

Comment:
i think this is OK for the back-compat reasons i mentioned on the other PR (if a bit odd) but we should take out the timeout argument to this function IMO if we do this (which does not appear to actually be used in any callsites outside of tests)